libmambapy/libmambapy/__init__.pyi

```diff
@@ -962,13 +962,15 @@ class Solver:
     def add_pin(self, arg0: str) -> None: ...
     def all_problems_structured(self) -> typing.List[mamba::MSolverProblem]: ...
     def all_problems_to_str(self) -> str: ...
+    def explain_problems(self) -> str: ...
     def is_solved(self) -> bool: ...
+    def must_solve(self) -> None: ...
     def problems_to_str(self) -> str: ...
     def set_flags(self, arg0: typing.List[typing.Tuple[int, int]]) -> None: ...
     def set_postsolve_flags(
         self, arg0: typing.List[typing.Tuple[int, int]]
     ) -> None: ...
-    def solve(self) -> bool: ...
+    def try_solve(self) -> bool: ...
```

Comment:
this is going to break all downstream projects. Maybe add a deprecation warning and a shim that forwards to either of the two functions?

Reply:
That's a good point! I did not do it at first because `Solver` was in `include/core`. Do we maintain API there as well?

Reply:
Conda seems to not use it directly
https://cs.github.com/?scopeName=All+repos&scope=&q=org%3Aconda+.solve%28%29

Reply:
This function is 100% used in boa, the conda-forge bot and the conda-libmamba-solver: https://github.com/conda-incubator/conda-libmamba-solver/blob/96e907cfbde41704c3fb50e59995d33c2d10359d/conda_libmamba_solver/solver.py#L321-L322

Reply:
Added it back in Python with a deprecation.